qiskit_nature/problems/second_quantization/lattice/models/ising_model.py

```diff
@@ -0,0 +1,143 @@
+            uniform_onsite_potential: The on-site potential. (or uniform_external_field)
+            onsite_interaction: The strength of the on-site interaction, the Ising model does not
+                have an on-site interaction.
+
+        Returns:
+            The Ising model with uniform parameters.
+
+        Raises:
+            Warning: If the on-site interaction is not None.
+        """
+        if onsite_interaction is not None:
+            raise Warning(
+                "The Ising model does not have on-site interactions. Provided onsite-interaction "
+                "parameter will be ignored."
+            )
+        graph = lattice.graph
+        for node_a, node_b, _ in graph.weighted_edge_list():
+            if node_a != node_b:
+                graph.update_edge(node_a, node_b, uniform_hopping)
+
+        for node_a in graph.node_indexes():
+            if graph.has_edge(node_a, node_a):
+                graph.update_edge(node_a, node_a, uniform_onsite_potential)
+            else:
+                graph.add_edge(node_a, node_a, uniform_onsite_potential)
+
+        return cls(Lattice(graph))
+
+    @classmethod
+    def from_parameters(
+        cls, hopping_matrix: np.ndarray, onsite_interaction: complex = None
+    ) -> "IsingModel":
+        """Return the Hamiltonian of the Ising model from the given hopping matrix.
+
+        Args:
+            hopping_matrix: A real or complex valued square symmetric matrix.
+            onsite_interaction: The strength of the on-site interaction, the Ising model does not
+                have an on-site interaction.
+
+        Returns:
+            IsingModel: The Ising model generated from the given hopping matrix.
+
+        Raises:
+            Warning: If the on-site interaction is not None.
+            ValueError: If the hopping matrix is not square matrix, it is invalid.
+        """
+        if onsite_interaction is not None:
+            raise Warning(
+                "The Ising model does not have on-site interactions. Provided onsite-interaction "
+                "parameter will be ignored."
+            )
+        # make a graph from the hopping matrix.
+        # This should be replaced by from_adjacency_matrix of retworkx.
+        shape = hopping_matrix.shape
+        if len(shape) == 2 and shape[0] == shape[1]:
+            graph = PyGraph(multigraph=False)
+            graph.add_nodes_from(range(shape[0]))
+            for source_index in range(shape[0]):
+                for target_index in range(source_index, shape[0]):
+                    weight = hopping_matrix[source_index, target_index]
+                    if not weight == 0.0:
+                        graph.add_edge(source_index, target_index, weight)
+            lattice = Lattice(graph)
+            return cls(lattice)
+        else:
+            raise ValueError(
+                f"Invalid shape of `hopping_matrix`, {shape},  is given."
+                "It must be a square matrix."
+            )
```

Comment:
As far as I can tell, this function is identical in the `FermiHubbardModel`. May I suggest that we move this entirely into `LatticeModel`?

Reply:
Yes, I've added the methods to `LatticeModel` and changed slightly the `__init__` signature so that it works for both classes.